js/products.js

```diff
@@ -0,0 +1,76 @@
+// find and show selected product
+
+let foundProduct = {};
+let totalWeight = null;
+let sumKcalProduct = null;
+let totalKkal = null;
+
+productSelectElement.addEventListener("input", (event) => {
+  foundProduct = products.find(
+    (product) => product.name === event.target.value
+  );
+  if (foundProduct) {
+    productElement.className = "active";
+    nameProductElement.textContent = foundProduct.name;
+    imageProductElement.src = foundProduct.imageUrl;
+    kcalProductElement.textContent = foundProduct.kcal;
+    proteinsProductElement.textContent = foundProduct.proteins;
+    fatsProductElement.textContent = foundProduct.fats;
+    carbohydratesProductElement.textContent = foundProduct.carbohydrates;
+    updateThis(productSelectElement, null);
+  }
+});
+
+weightProductSelectElement.addEventListener("input", (event) => {
+  const productElementWeight = event.target.value / 100;
+  sumKcalProduct = productElementWeight * foundProduct.kcal;
+  let sumProteinsProduct = productElementWeight * foundProduct.proteins;
+  let sumFatsProduct = productElementWeight * foundProduct.fats;
```

Comment:
Move this calculation into separate function

function calculateNutrientsWeight(weight, product) {
  return {
     proteins: weight * product.proteins,
     fats: weight * product.fats,
     carbohydrates: weight * product.carbohydrates,
     kcal: weight * product.kcal,
  }
}


Now you can calculate the product nutrients with the following way:

const productWeight = event.target.value / 100;
const { proteins, fats, carbohydrates, kcal } = calculateNutrientsWeight(productWeight, foundProduct);